lib/radiator_web/resolvers/directory.ex

```diff
@@ -69,6 +69,22 @@ defmodule RadiatorWeb.Resolvers.Directory do
       nil ->
         {:error, "Podcast ID #{id} not found"}
 
+      %Podcast{slug: nil} = podcast ->
+        slug = Slugger.slugify_downcase(podcast.title)
+
+        attrs = %{
+          published_at: DateTime.utc_now(),
+          slug: slug
+        }
+
+        case Directory.update_podcast(podcast, attrs) do
+          {:ok, podcast} ->
+            {:ok, podcast}
+
+          {:error, %Ecto.Changeset{errors: [slug: {"has already been taken", _}]}} ->
+            {:error, "Slug #{slug} has already been taken"}
```

Comment:
I think we should try to generate unique slugs and not reject them, meaning: saving of a podcast (or any other sluggable entity) should never fail due to a unique slug constraint.

I suggest copying what WordPress does: appending an incrementing number until a unique slug is found. `foo-bar` -> `foo-bar-1` -> `foo-bar-2` etc.